app/src/main/java/com/example/drawingApp/DialogUtility.kt

```diff
@@ -0,0 +1,157 @@
+package com.example.drawingApp
+
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+object DialogUtility {
+
+    const val MAX_CHAR_LEN = 16
+
+    /**
+     * used to updated the sheet adapter with the onEdit lambda
+     * the sheet is for dismissing to avoid window leaks
+     */
+    data class SheetObject(val onEdit: (Int, String) -> Unit, val sheet: BottomSheetDialog)
+
+    /**
+     * opens a dialog that will take a string and allows you to interact with the string
+     * returns the string to wherever on the onSubmission
+     */
+    fun showLayerAlertDialog(
+        context: Context,
+        startString: String = "",
+        isCreating: Boolean = true,
```

Comment:
`isCreatingNewLayer`? Hard to look at this and tell what it does and there's no kdoc

Reply:
isCreating is to tell what the dialog is used for if it's for creating a new layer or editing one. This is purely used to update the title on the dialog.

Reply:
updated the name as well